core/src/main/scala/za/co/absa/spline/harvester/postprocessing/extra/ExtraMetadataCapturingFilter.scala

```diff
@@ -19,74 +19,103 @@ package za.co.absa.spline.harvester.postprocessing.extra
 import org.apache.commons.configuration.Configuration
 import org.apache.commons.io.IOUtils
 import org.apache.spark.internal.Logging
-import za.co.absa.commons.config.ConfigurationImplicits.ConfigurationOptionalWrapper
+import za.co.absa.commons.config.ConfigurationImplicits.ConfigurationRequiredWrapper
 import za.co.absa.spline.harvester.ExtraMetadataImplicits._
 import za.co.absa.spline.harvester.HarvestingContext
 import za.co.absa.spline.harvester.json.HarvesterJsonSerDe.impl._
 import za.co.absa.spline.harvester.postprocessing.PostProcessingFilter
-import za.co.absa.spline.harvester.postprocessing.extra.DeclarativeExtraInjectingFilter.ExtraDef
-import za.co.absa.spline.harvester.postprocessing.extra.model.predicate.{BaseNodeNames, Predicate}
+import za.co.absa.spline.harvester.postprocessing.extra.ExtraMetadataCapturingFilter.{ExtraDef, createDefs, evaluateExtraDefs}
+import za.co.absa.spline.harvester.postprocessing.extra.model.predicate.{BaseNodeName, Predicate}
 import za.co.absa.spline.harvester.postprocessing.extra.model.template.ExtraTemplate
 import za.co.absa.spline.producer.model.v1_1._
 
 import java.net.URL
+import scala.util.Try
 
-class DeclarativeExtraInjectingFilter(
-  planExtraDefs: Seq[ExtraDef],
-  eventExtraDefs: Seq[ExtraDef],
-  readExtraDefs: Seq[ExtraDef],
-  writeExtraDefs: Seq[ExtraDef],
-  operationExtraDefs: Seq[ExtraDef]
-) extends PostProcessingFilter {
+class ExtraMetadataCapturingFilter(allDefs: Map[BaseNodeName.Value, Seq[ExtraDef]]) extends PostProcessingFilter {
```

Comment:
If I want to just add some tags to the lineage, I can specify a rule to do that. In that case I do not capture anything, I am just **injecting** the extra tags.

In case I want to capture additional information from lineage, I can specify the rule to **captrue** it and than **injecting** as an extra data to lineage.

My point is the capturing happens only in some cases, but the extra data are always injected into the standard lineage, that is the purpose of the filter.

Reply:
So, if I get it right we are merely talking about wording here. From what you describe I suggest the the word  '__collecting__'. The problem with the word '__injecting__' is that it puts too much emphasis on what it does with the extra metadata (injects it somewhere) rather than focusing on the primary goal which is to collect (capture) some extra stuff in addition to what is collected by default.